### DISCUSSION TITLE
New capability is required for MSIX install

### DISCUSSION THREAD

**Discussion by alippai** [microsoft/terminal#16668] [Q&A] [answered]
### Steps to reproduce

With the recent version 1.19.10302.0 the MSIX installer needs new capabilities it didn't need before.

### Expected Behavior

No extra capabilities install possible

### Actual Behavior

Capability: `Write registry entries and files that are not cleaned up on install` is required

**Comment by zadjii-msft** [comment]
That was added in c7816fdb05338317cbfa52620171b7510400690c, which dates back to at least [v1.18.1421.0](https://github.com/microsoft/terminal/releases/tag/v1.18.1421.0). 

Pretty sure this is entirely by design. Is this an issue for some reason/?

**Comment by alippai** [comment]
Thanks for the swift answer. I came across it because in some environments that usage might be restricted.
I'm not sure what the exact implementation is, but the capability description implies data is left around even after uninstall which sounds less streamlined than the previous version. Is there a way to tie that file/registry writing to the installation instead?

**Comment by DHowett** [comment, marked as answer]
> the capability description implies data is left around even after uninstall which sounds less streamlined than the previous version

This was actually always true, it was just never declared.

The registry key that Windows Terminal can write to is `HKCU\Console\%%Startup`. It is how we communicate with the rest of the console subsystem that a given version of Terminal should be used as the host for console applications.

It supports this feature:

![image](https://github.com/microsoft/terminal/assets/189190/1fd790e8-b908-4623-8d20-e269a404fcea)

If stale data is left in that key and the Terminal uninstalled, the system will fall back to the "Windows Console Host" option.
